lms/djangoapps/teams/tests/test_views.py

```diff
@@ -1769,6 +2050,59 @@ def test_expand_team(self):
         )
         self.verify_expanded_team(result['team'])
 
+    @ddt.data(
+        ('student_masters', 200),
+        ('student_masters_not_on_team', 200),
+        ('student_unenrolled', 404),
+        ('student_enrolled', 403),
+        ('student_enrolled_both_courses_other_team', 403),
+        ('staff', 200),
+    )
+    @ddt.unpack
+    def test_organization_protected(self, user, expected_status):
+        """ Users should not be able to see memberships for users in a different bubble than them """
+        self.get_membership_detail(
+            self.masters_only_team.team_id,
+            self.users['student_masters'].username,
+            expected_status,
+            user=user
+        )
+
+    @ddt.unpack
+    @ddt.data(
+        ('student_on_team_1_private_set_1', 200),
+        ('student_unenrolled', 404),
+        ('student_enrolled', 403),
+        ('student_on_team_2_private_set_1', 403),
+        ('staff', 200)
+    )
+    def test_private_managed_team(self, user, expected_status):
+        """
+        Users should not be able to see memberships for users in private_managed
```

Comment:
student_enrolled and sot2ps1 should not be able to tell that there's a detail that they don't have access to